### DISCUSSION TITLE
In analogy of spring-kafka-test spring-pulsar should expose a similar artifact

### DISCUSSION THREAD

**Discussion by jonas-grgt** [spring-projects/spring-pulsar#552] [Ideas]
I was surprised to see that spring-kafka-test, although only containing the minimal `PulsarTestContainerSupport` is not released.

I think the spring-pulsar community could benefit from a test project that incorporates TestCunsumer utlities, hamcrest or AssertJ assertions etc ...

**Comment by onobc** [comment]
`s/spring-kafka-test/spring-pulsar-test`

Agreed @jonasgeiregat . The intention is to release once we have some of the consumer test facilities stand out that could be useful. If you have some suggestions in this area about what to expose, we can work towards that together. 

Thanks

**Reply by jonas-grgt** [reply]
One idea I had was something similar to `KafkaTestUtils` that can be useful for integration tests that would include functionality to fetch Messages, like:

- getMessages(Consumer<T> consumer)
- awaitUntillMessage(Consumer<T> consumer, Predicate<T> condition, Duration timeout)
- etc ..

Another was adding custom AssertJ `Conditions`.

**Reply by onobc** [reply]
That sounds like a good first set of functionality. That paired w/ the PulsarContainerSupport would be valuable IMO. 

Would you be willing to submit a PR to add some of the above utilities into the existing test module?

**Reply by jonas-grgt** [reply]
I would love to. But, simply adding them to the testing module wouldn't be enough as the module itself is not published, is it?

**Reply by onobc** [reply]
Yep, I can handle the part to make it publish. :)

**Comment by onobc** [comment]
Doing a bit of GH discussions housekeeping...

Since @jonas-grgt has so kindly contributed this idea starting w/ `1.1.0` -  going to close the discussion as resolved.

Thanks @jonas-grgt for the discussion and the following contributions.
